clusterloader2/pkg/measurement/common/metrics_server_prometheus.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+	"time"
+)
+
+const (
+	metricsServerPrometheusMeasurementName = "MetricsServerPrometheus"
+
+	metricsServerLatencyQuery = "apiserver_request_duration_seconds_bucket{group=metrics.k8s.io,resource=pods,scope=cluster}"
+)
+
+func init() {
+	create := func() measurement.Measurement { return CreatePrometheusMeasurement(&metricsServerGather{}) }
+	if err := measurement.Register(metricsServerPrometheusMeasurementName, create); err != nil {
+		klog.Fatalf("Cannot register %s: %v", metricsServerPrometheusMeasurementName, err)
+	}
+}
+
+type metricsServerGather struct{}
+
+func (g *metricsServerGather) Gather(executor QueryExecutor, startTime time.Time, config *measurement.Config) ([]measurement.Summary, error) {
+	finalLatency, err := g.gatherLatencySummary(executor, startTime)
+	if err != nil {
+		return nil, err
+	}
+
+	// Pretty Print the report.
+	result := &measurementutil.LatencyMetric{}
+	if err = SetQuantileFromHistogram(result, finalLatency); err != nil {
+		return nil, err
+	}
+	content, err := util.PrettyPrintJSON(result)
+	if err != nil {
+		return nil, err
+	}
+	// Create Summary.
+	return []measurement.Summary{measurement.CreateSummary(ksmLatencyName, "json", content)}, nil
+}
+
+func (g *metricsServerGather) IsEnabled(config *measurement.Config) bool {
+	return config.CloudProvider.Features().SupportMetricsServerMetrics
+}
+
+func (g *metricsServerGather) String() string {
+	return metricsServerPrometheusMeasurementName
+}
+
+func (g *metricsServerGather) gatherLatencySummary(executor QueryExecutor, startTime time.Time) (*measurementutil.Histogram, error) {
```

Comment:
Unless I am missing something, I think you could ski this function.
Please take a look at https://github.com/kubernetes/perf-tests/blob/81c96c34e3c1f11c5fe91744ef7ce7bf44c2fe5c/clusterloader2/pkg/measurement/common/nodelocaldns_latency_prometheus.go#L32
Check how percLatencyQueryTemplate looks like and how it's used. It computes latencies for specific period of time (so you don't need to do this computation on histograms)